contracts/fee/FeeHandler.sol

```diff
@@ -3,46 +3,176 @@
 pragma solidity ^0.8.0;
 
 import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 
 import "../data/DataStore.sol";
 import "../role/RoleModule.sol";
+import "../utils/BasicMulticall.sol";
 import "../fee/FeeUtils.sol";
+import "../data/Keys.sol";
+import "../v1/IVaultV1.sol";
+import "./IFeedAddress.sol";
 
 // @title FeeHandler
-contract FeeHandler is ReentrancyGuard, RoleModule {
+contract FeeHandler is ReentrancyGuard, RoleModule, BasicMulticall {
+    using SafeERC20 for IERC20;
+
     DataStore public immutable dataStore;
     EventEmitter public immutable eventEmitter;
+    address public immutable vaultV1;
+    address public immutable gmx;
 
     constructor(
         RoleStore _roleStore,
         DataStore _dataStore,
-        EventEmitter _eventEmitter
+        EventEmitter _eventEmitter,
+        address _vaultV1,
+        address _gmx
     ) RoleModule(_roleStore) {
         dataStore = _dataStore;
         eventEmitter = _eventEmitter;
+        vaultV1 = _vaultV1;
+        gmx = _gmx;
+    }
+
+    // @dev withdraw fees from this contract
+    // @param token the token for which to claim fees
+    // @param amount the amount of fees to claim
+    function withdrawFees(address token, uint256 amount) external nonReentrant onlyFeeKeeper {
+        address receiver = dataStore.getAddress(Keys.FEE_RECEIVER);
```

Comment:
should allow "markets[]" to be passed in and call FeeUtils.claimFees(dataStore, eventEmitter, market, token, address(this)), to allow all WNT or GMX to be directly claimed from the markets

Reply:
withdrawFees has now been updated to include an address[] calldata markets input param. Currently, the buybackToken fees are first consolidated in FeeHandler and then transferred to FEE_RECEIVER. If preferred, the claimed fees could be transferred directly to FEE_RECEIVER from the GM market. 

Also, this could be updated to also include an array of buybackTokens to allow for both GMX and WNT to be withdrawn in one call, though I think this would require either matching the length of markets and buybackTokens or a nested for loop.

Reply:
the current logic looks ok to me